### ISSUE TITLE
Virtual workspaces collapse into one after suspend

### ISSUE THREAD

**Issue by danielbernalo** [danielbernalo/gnome-workspace-islands#7]
## What happened

Arrange windows across the virtual workspaces of a secondary monitor, suspend the machine, resume and unlock. Every window comes back piled into a **single** virtual workspace instead of spread across the ones they were on.

The windows do return to the secondary monitor — it is the grouping that is lost, not the placement.

## GNOME Shell version

GNOME Shell 50.3

## Session type

Wayland

## Monitor setup

Two: laptop panel as primary, one external display as secondary.

## Root cause

Three independent causes, all verified against the shell's own JS:

**1. The window → workspace mapping is destroyed and then guessed back.**
`disable()` and `MonitorState.detach()` both clear `_groups`, leaving only `activeIndex` and the WM_CLASS app rules. `track()` then rebuilds membership from `indexForApp()`, falling back to `activeIndex`. Since `rememberApp()` only runs when the user moves a window by hand, a window never explicitly moved has no rule at all and lands on `activeIndex`. Every app ends up in one place.

The information was recoverable the whole time: `persistence.js` is right that windows have no identity *across sessions*, but a lock, an unplug and a suspend all happen inside one, where the `Meta.Window` objects are the same objects.

**2. `disable()` runs on every lock, and `session-modes` alone cannot stop it.**
`metadata.json` declares no `session-modes`, so the lock screen disables the extension (`lock-enabled=true`, `lock-delay=0` — suspend locks immediately). But adding it is not sufficient: `ExtensionManager._callExtensionDisable` rebases the extension order, so disabling *any* extension also disables and re-enables every extension after it. With six `user`-only extensions ahead of this one in `enabled-extensions`, every lock tears it down six times over.

**3. `connectorOf()` is dead code.**
It reads `monitor.connector`, but the shell's `Monitor` is `(index, geometry, scale)` and has no such property — the string does not appear in `layout.js` at all. So every key, in the registry and in `saved-state`, was `monitor-${index}`: exactly what the module header says must be avoided, because indices are reshuffled on hotplug.

## Journal output

```
workspace-islands: enabled — 1 secondary monitor(s), workspaces-only-on-primary=true
workspace-islands: enabled — 1 secondary monitor(s), workspaces-only-on-primary=true
workspace-islands: enabled — 1 secondary monitor(s), workspaces-only-on-primary=true
```

Repeated enable lines are cause 2 in the log: the extension being rebuilt from scratch on every lock.

## Other enabled extensions

advanced-weather@sanjai.com, clipboard-history@alexsaveau.dev, azwallpaper@azwallpaper.gitlab.com, auto-accent-colour@Wartybix, status-tray@keithvassallo.com, Vitals@CoreCoding.com

All six are `session-modes: [user]` and all six sit ahead of this extension in `enabled-extensions`, which is what makes cause 2 fire on every lock.

## Preconditions

- [x] `workspaces-only-on-primary` is `true`
- [x] PaperWM is not enabled
- [x] I still see the problem with **Slide animation** turned off
